src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -514,19 +533,78 @@ public Task<bool> StopProjectAsync(CancellationToken cancellationToken)
             return _sessionManager.StopProjectAsync(this, cancellationToken).AsTask();
         }
 
-        public Task<bool> RestartProjectAsync(CancellationToken cancellationToken)
+        public IDeltaApplier? GetDeltaApplier()
+        {
+            return ProjectHotReloadSessionManager.GetDeltaApplier(this);
+        }
+    }
+
+    private class SolutionBuildCompleteListener : IVsUpdateSolutionEvents, IDisposable
+    {
+        private readonly TaskCompletionSource<bool> _buildCompletedSource = new();
+
+        public SolutionBuildCompleteListener()
         {
-            return TaskResult.False;
         }
 
-        public Task<bool> RestartProjectAsync(bool isRunningUnderDebug, CancellationToken cancellationToken)
+        public static int UpdateSolution_Start(ref int pfCancelUpdate)
         {
-            return _sessionManager.RestartProjectAsync(this, isRunningUnderDebug, cancellationToken);
+            return HResult.OK;
+        }
+        public static int UpdateSolution_Progress(ref int pfCancelUpdate)
+        {
+            return HResult.OK;
         }
```

Comment:
Are these static methods used? They look like interface methods, but cannot be if they're static. I can't see any usages. Remove them?

Reply:
No they are not used anywhere, will remove them